gridData/tests/test_dx.py

```diff
@@ -10,19 +10,9 @@
 
 from . import datafiles
 
-def test_read_dx():
-    g = Grid(datafiles.DX)
-    POINTS = 8
-    ref = np.ones(POINTS)
-    ref[4] = 1e-6
-    ref[5] = -1e+6
-    assert_equal(g.grid.flat, ref)
-    assert_equal(g.grid.size, POINTS)
-    assert_equal(g.delta, np.ones(3))
-    assert_equal(g.origin, np.array([20.1, 3., -10.]))
-
-def test_read_dxgz():
-    g = Grid(datafiles.DXGZ, file_format='DXGZ')
+@pytest.mark.parametrize("infile", [datafiles.DX, datafiles.DX+'.gz'])
```

Comment:
I think you also need to
- add a `test.dx.gz` to test files
- add `DXGZ` to `datafiles/__init__.py`

I don't know if one can make a fixture work in which you generate the gzipped file on the fly. I can think of the first part, but not how to use the fixture together with `datafiles.DX` in a parameterized test or fixture. But in any case, here's the code I can think of (uses [tmpdir_factory](https://docs.pytest.org/en/latest/tmpdir.html#the-tmpdir-factory-fixture)  because the temporary file should be created in a temporary location but the `tmpfile` fixture is not module level)
```python
@pytest.fixture(scope="module")
def dxgz_file(tmpdir_factory, src=datafiles.DX):
    basename = os.path.basename(src) + '.gz'
    fn = tmpdir_factory.mktemp("compressed").join(basename)
    with open(src, 'rb') as inp:
        with gzip.open(str(fn), "wb") as out:
             out.write(inp.read())
    return str(fn)

# use dxgz_file fixture when needed ... although I am actually not sure how 
# you use a fixture as part of a parameterized fixture.
```

Reply:
I added test.dx.gz file and DXGZ to datafiles/__init__.py but I'm not sure about the fixture...